scripts/xtensa-build-zephyr.sh

```diff
@@ -186,7 +186,8 @@ build()
 			/bin/pwd
 			set -x
 			west build --build-dir "$bdir" --board "$PLAT_CONFIG" \
-				zephyr/samples/subsys/audio/sof
+				zephyr/samples/subsys/audio/sof \
+				-- -DEXTRA_CFLAGS='-Werror'
```

Comment:
it begs the question why we are not enabling -Werror for the regular firmware. I thought it was the case....

Reply:
We do have `-Werr` in the XTOS build. It's the `__attribute__ ((format (printf, x, y)));` that we don't have in the XTOS build. I think I tried once and realized it was not possible because of the complex dictionary macros, not sure.

Reply:
@marc-hb see https://github.com/thesofproject/sof/pull/4650/, there was no error reported for the code you fixed. That makes me wonder why the XTOS build succeeded at all?

Reply:
This warning was reported in #4650, see the log of the Zephyr build athttps://github.com/thesofproject/sof/runs/3504732756

However github (or Jenkins, etc.) have no concept of warning, it's either red or green but never yellow. That's why this PR adds `-Werror`.

I saw the warning when building Zephyr locally because I look at the pretty colors.

The logger in XTOS has no `__attribute__ ((format (printf, x, y)));` so no warning or error or anything.

Reply:
Yep, the Cadence xtos code is not compiled with newer GCCs by Cadence so will through up warnings and they are aware of this for future versions and updates (at this point -Werror can be global).
We also need to use ```--std=c99``` globally..